installer/install.sh.in

```diff
@@ -1,234 +1,9 @@
-_err_exit $? "Failed to activate virtual evironment "$ROOTDIR"/.venv. Please see $TROUBLESHOOTING for help."
-
-PYTHON="$ROOTDIR"/.venv/bin/python
-$PYTHON -mensurepip --upgrade
-$PYTHON -mpip install --upgrade pip
-
-#--------------------------------------------------------------------------------
-echo
-echo "*** Installing InvokeAI Dependencies ***"
-
-if [ "$OS_NAME" == "osx" ]; then
-    echo "macOS detected. Installing MPS and CPU support."
-    egrep -v '^-e .' environments-and-requirements/requirements-mac-mps-cpu.txt >requirements.txt
-else
-    if (lsmod | grep amdgpu) &>/dev/null ; then
-	readinput -e -p "Linux system with AMD GPU driver detected. Install ROCm AMD accelerated support? (Otherwise, CUDA support will be installed) [n]: " input
-	RESPONSE=${input:='n'}
-	if [ "$RESPONSE" != "n" ]; then
-	    echo "Installing ROCm (AMD) support"
-	    egrep -v '^-e .' environments-and-requirements/requirements-lin-amd.txt >requirements.txt
-	else
-	    echo "Installing CUDA support"
-	    egrep -v '^-e .' environments-and-requirements/requirements-lin-cuda.txt >requirements.txt
-	fi
-    else
-	echo "Linux system detected. Installing CUDA and CPU support."
-	egrep -v '^-e .' environments-and-requirements/requirements-lin-cuda.txt >requirements.txt
-    fi
-fi
-
-$PYTHON -mpip install --prefer-binary -r requirements.txt
-_err_exit $? "Failed to install InvokeAI's dependencies."
-
-#--------------------------------------------------------------------------------
-echo
-echo "*** Installing InvokeAI Modules and Executables ***"
-$PYTHON -mpip install $INVOKE_AI_SRC
-_err_exit $? "Installation of InvokeAI failed."
-
-#--------------------------------------------------------------------------------
-echo " *** Setting Up Root Directory "$ROOTDIR" *** "
-cp -pr templates/rootdir/* "$ROOTDIR"/
-cp templates/invoke.sh.in "$ROOTDIR"/invoke.sh
-chmod a+rx "$ROOTDIR"/invoke.sh
-cp templates/update.sh.in "$ROOTDIR"/update.sh
-chmod a+rx "$ROOTDIR"/update.sh
-
-# This allows the updater to work!
-cp -pr environments-and-requirements requirements.txt "$ROOTDIR/"
-
-#--------------------------------------------------------------------------------
-echo
-echo "*** Confguring InvokeAI ***"
-pushd "$ROOTDIR" >/dev/null
-$PYTHON ./.venv/bin/configure_invokeai.py --root="$ROOTDIR"
-_err_exit $? "Initial configuration failed. Please see above error messages and $TROUBLESHOOTING for help."
-
-#--------------------------------------------------------------------------------
-popd
-cp templates/invoke.sh.in "$ROOTDIR"/invoke.sh
-chmod a+rx "$ROOTDIR"/invoke.sh
-
-cp templates/update.sh.in "$ROOTDIR"/update.sh
-chmod a+rx "$ROOTDIR"/update.sh
-
-echo "You may now run InvokeAI by entering the directory $ROOTDIR and running invoke.sh:"
-echo
-echo "  ${ROOTDIR}/invoke.sh"
-echo
 
-read -e -p "Run InvokeAI now? [y]:" input
-RESPONSE=${input:='y'}
-if [ "$RESPONSE" == 'y' ]; then
-    exec ${ROOTDIR}/invoke.sh
-fi
+exec python3 $(dirname $0)/main.py ${@}
```

Comment:
This is not working if no argument is given, this fixes it:

```sh
exec python3 $(dirname $0)/main.py ${1:+${@}}
```

Reply:
Interesting, this is on macOS I assume? What is your `bash` version?

Reply:
ah, silly me. that `set -u` is unnecessary. in fact, all of those shell options are unnecessary. I remove them in the latest commit.